src/session/session_api.c

```diff
@@ -1594,6 +1594,27 @@ __session_salvage_readonly(WT_SESSION *wt_session, const char *uri, const char *
     API_END_RET(session, ret);
 }
 
+/*
+ * __check_key_outside_truncate_range --
+ *     Checks if the key falls outside the truncate range, inclusive of the start and stop keys if
+ *     provided.
+ */
+static bool
+__check_key_outside_truncate_range(WT_CURSOR *current, WT_TRUNCATE_INFO *trunc_info)
+{
+    if (trunc_info->orig_start_key != NULL) {
+        if (current->key.data < trunc_info->orig_start_key->data)
```

Comment:
You should use the key comparison function (`__wt_compare`) instead of comparing `data` directly (as `data` is a pointer).

Reply:
This is a good suggestion, but I'm having trouble accessing the collator needed for the compare function as we don't have the handle. I'll see if there's a workaround.